### PR TITLE
Add `ticl_v5` modifier at HLT

### PR DESCRIPTION
**Pull request by AuroraPerego**
#### PR description:

This PR extends the existing `ticl_v5` modifier to the Phase-2 HLT. The default for now remains TICLv4.
In all the `.203` wfs the `--procModifiers ticl_v5` option is added automatically.
The changes follow the logic of the offline workflow, but with the MTD timing disabled by default.

#### PR validation:

Checked on wf `29688.203`. 
**A workflow `.203` (e.g. `29688.203` and `29888.203`) with the option `-w upgrade` should be used to test this PR.**

FYI @waredjeb

### CODE REVIEW

**Comment by felicepantaleo** [comment]
test parameters:  

- workflow_opts= -w upgrade
- workflow = 29888.203,29688.203,29846.204,29846.205,29891.203,29891.204,29891.205

**Comment by felicepantaleo** [comment]
@cmsbuild please test

File: HLTrigger/Configuration/python/HLT_75e33/modules/filteredLayerClustersPassThrough_cfi.py
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/modules/filteredLayerClustersPassThrough_cfi.py:3]: This should start with `hlt`

File: HLTrigger/Configuration/python/HLT_75e33/modules/ticlCandidate_cfi.py
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/modules/ticlCandidate_cfi.py:3]: This should start with `hlt`

File: HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersPassthrough_cfi.py
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersPassthrough_cfi.py:3]: This should start with `hlt`

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by felicepantaleo** [comment]
test parameters:  

- workflow_opts= -w upgrade
- workflow = 29888.203,29688.203

**Comment by felicepantaleo** [comment]
@cmsbuild please test

File: HLTrigger/Configuration/python/HLT_75e33/sequences/ticlCandidateSequence_cfi.py
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/sequences/ticlCandidateSequence_cfi.py:5]: Sorry, I missed this earlier. Seqences run at HLT start their label with `HLT` - please do so also for Phase-2, here and elsewhere.

File: HLTrigger/Configuration/python/HLT_75e33/sequences/iterTICLSequence_cfi.py
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/sequences/iterTICLSequence_cfi.py:11]: This predates this PR, but please fix it here as well (would be nice to fix all of them, but perhaps outside the scope of your PR).

**Review by mmusich — CHANGES_REQUESTED** [review]

File: HLTrigger/Configuration/python/HLT_75e33/sequences/iterTICLSequence_cfi.py
Comment (AuroraPerego) [reply HLTrigger/Configuration/python/HLT_75e33/sequences/iterTICLSequence_cfi.py:11]: ok no problem (for the non-HGCAL modules/sequences I'll see what I manage to do)

**Comment by mandrenguyen** [comment]
please test

**Comment by mmusich** [comment]
@AuroraPerego many thanks for providing the renaming of the HLT modules and sequences in the phase2 menu! 
I would be in favor though to descope such a large commit to another PR, considering also that all downstream consumers (e. g. dqm and validation) would need to adjust to consume the new product labels (see e.g. empty plots in the [DQM bin by bin comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-07-20-1100+b7fb03/63676/dqm-histo-comparison-summary.html)). 
Would you mind rolling back that last commit? We can make a joint PR soon thereafter using it from the TSG side.

**Comment by AuroraPerego** [comment]
> @AuroraPerego many thanks for providing the renaming of the HLT modules and sequences in the phase2 menu! I would be in favor though to descope such a large commit to another PR, considering also that all downstream consumers (e. g. dqm and validation) would need to adjust to consume the new product labels (see e.g. empty plots in the [DQM bin by bin comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-07-20-1100+b7fb03/63676/dqm-histo-comparison-summary.html)). Would you mind rolling back that last commit? We can make a joint PR soon thereafter using it from the TSG side.

Sure, no problem!
Just to confirm with you before doing it, I'll leave the `hlt` label in front of the new modules I've added, but I won't add it in front of already existing HGCAL/TICL modules and leave that to the PR with all the other renaming.

**Comment by AuroraPerego** [comment]
> Perfect, thanks a lot and sorry for the back and forth (please preserve the commit in a separate branch so we can cherry-pick later!)

To not forget where it is: [TICLv5HLT_fix45426](https://github.com/AuroraPerego/cmssw/tree/TICLv5HLT_fix45426)@[e638a3f](https://github.com/AuroraPerego/cmssw/commit/e638a3fa1ab8db6d1862ba35bca959a84fd26f6b)

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by felicepantaleo** [comment]
@cmsbuild please test

**Comment by felicepantaleo** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@AuroraPerego @felicepantaleo overall I don't have further comments on the naming etc.
I just notice that we have a drop of efficiency in the [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-07-22-1100+b7fb03/63685/triggerResults): for `.203` workflows

```
      Events    Accepted      Gained        Lost       Other  Trigger
          10           0           -           -          ~1  HLT_DoubleMediumChargedIsoPFTauHPS40_eta2p1
          10           0           -           -          ~1  HLT_DoubleMediumDeepTauPFTauHPS35_eta2p1
          10          10           -           -         ~10  MC_JME
          10          10           -           -         ~10  MC_BTV
          10           2           -          -2           -  MC_Ele5_Open_Unseeded
          10           2           -          -2           -  MC_Ele5_Open_L1Seeded
```
Can you comment if you expect loosing events in the `MC_Ele5_Open*` paths?

**Comment by felicepantaleo** [comment]
@mmusich it is expected, as the pidcut is disabled here (hopefully the link works, I'm from the phone):
https://github.com/cms-sw/cmssw/pull/45500/files?file-filters%5B%5D=.py&show-viewed-files=true#diff-603a57a6960d020c329208a6fbcf0d0b267a03af039ccbebccc447743a58e697

it will be reenabled in another PR in which we update the pid model

**Comment by mmusich** [comment]
+hlt
   * for https://github.com/AuroraPerego/cmssw/commit/6633fcf7ecacd555db6e06937b028f74b06bbd4d
   *  extends existing `ticl_v5` modifier to the Phase-2 HLT. 
   * change in the `TriggerResults` understood as in https://github.com/cms-sw/cmssw/pull/45500#issuecomment-2246228641

**Comment by mandrenguyen** [comment]
I was surprised to see such a big change in the Particle ID [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-07-22-1100+b7fb03/63685/validateJR/29688.203_SinglePiPt25Eta1p7_2p7+2026D110_ticl_v5/all_HLT_step2/c_recoPFCandidates_pfTICL__HLT_obj_particleId.png
)
Is that expected?

**Comment by AuroraPerego** [comment]
> I was surprised to see such a big change in the Particle ID [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-07-22-1100+b7fb03/63685/validateJR/29688.203_SinglePiPt25Eta1p7_2p7+2026D110_ticl_v5/all_HLT_step2/c_recoPFCandidates_pfTICL__HLT_obj_particleId.png) Is that expected?

Yes, part of the changes in TICLv5 affects the linking in the calorimeter and with the track. Both have been improved, so now we are creating fewer PF candidates, particularly less neutral ones.

**Comment by mmusich** [comment]
upon second look, there is some collection which is empty now:

image

or reduced in content:

image

**Comment by mmusich** [comment]
> Both have been improved, so now we are creating fewer PF candidates, particularly less neutral ones.

OK so I guess this explains the reduction in `reco::PFCandidates`. What about `ticlTrackstersMerge`?

**Comment by AuroraPerego** [comment]
> OK so I guess this explains the reduction in `reco::PFCandidates`. What about `ticlTrackstersMerge`?

They do not exist anymore in TICLv5 because they have been replaced with `ticlCandiates`.

**Comment by mandrenguyen** [comment]
+reconstruction
Changes expected, as explained in [comment](https://github.com/cms-sw/cmssw/pull/45500#issuecomment-2248599496)

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by mandrenguyen** [comment]
[AuroraPerego](https://github.com/AuroraPerego) can you please resolve addressed open conversations while we wait for @cms-sw/upgrade-l2  to review/sign?

**Comment by AuroraPerego** [comment]
> [AuroraPerego](https://github.com/AuroraPerego) can you please resolve addressed open conversations while we wait for @cms-sw/upgrade-l2 to review/sign?

done :)

**Comment by srimanob** [comment]
+Upgrade

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**
